tests/cypress/e2e/frontend/settingsPreferencesPage.cy.ts

```diff
@@ -80,277 +81,381 @@ describe('Settings - preferences page', () => {
-      login();
+              cy.get('[aria-label="Compare now"').first().click();
+              cy.get('button#expert_submit_btn').click();
 
-      cy.get('[data-testid=videos_recommendations__default_unsafe]').click();
-      cy.contains('Update preferences').click();
+              cy.contains('test_exclude_false').click();
+              cy.get('[data-testid="settings-preferences"]').click();
+              cy.wait('@settingsRetrievedFromAPI');
 
-      cy.get('a[aria-label="Link to the recommendations page"]').should(
-        'have.attr', 'href', '/recommendations?date=Month&advanced=unsafe&language=en'
-      );
-    });
-  });
+              // Change an additional setting to bypass the cache in /feed/foryou.
+              cy.get('[data-testid="videos_feed_foryou__unsafe"]').check();
+              cy.contains('Update preferences').click();
 
-  describe('Setting - exclude compared entities', () => {
-    it('handles the value false (exclude)', () => {
-      cy.visit('/settings/preferences');
-      login();
+              cy.visit('/feed/foryou');
+              cy.wait('@recommendationsRetrievedFromAPI');
 
-      cy.get('[data-testid=videos_recommendations__default_exclude_compared_entities]');
-      cy.contains('Update preferences').click();
+              cy.get('[data-testid="video-card-info"] h5')
+                .first()
+                .invoke('attr', 'title')
+                .should('eq', videoTitle);
+            });
 
-      cy.get('a[aria-label="Link to the recommendations page"]').should(
-        'have.attr', 'href', '/recommendations?date=Month&language=en'
-      );
-    });
+            cy.deleteUser('test_exclude_false');
+        });
 
-    it('handles the value true (include)', () => {
-      cy.visit('/settings/preferences');
-      login();
+        it('handles the value true (exclude)', () => {
+          cy.recreateUser('test_exclude_true', 'test_exclude_true@example.com', 'tournesol');
 
-      cy.get('[data-testid=videos_recommendations__default_exclude_compared_entities]')
-        .click();
-      cy.contains('Update preferences').click();
+          cy.intercept('http://localhost:8000/polls/videos/recommendations/*')
+            .as('recommendationsRetrievedFromAPI');
 
-      cy.get('a[aria-label="Link to the recommendations page"]').should(
-        'have.attr', 'href', '/recommendations?date=Month&advanced=exclude_compared&language=en'
-      );
+          cy.intercept('http://localhost:8000/users/me/settings/')
+            .as('settingsRetrievedFromAPI');
+
+          cy.visit('/login');
+          login('test_exclude_true');
+          cy.contains('test_exclude_true').click();
+          cy.get('[data-testid="settings-preferences"]').click();
+
+          cy.wait('@settingsRetrievedFromAPI');
+          cy.get(videosForYouDateSelector).click();
+          cy.contains('All time').click();
+          cy.get('[data-testid=videos_feed_foryou__exclude_compared_entities]');
```

Comment:
If I understand correctly, we expect the checkbox to be checked by default:
```suggestion
          cy.get('[data-testid=videos_feed_foryou__exclude_compared_entities]').should('be.checked');
```

Reply:
Fixed by 9ebc7272025525d92a700a941b19a31136cb0550